add placement strategy (#385)
open-cluster-management-io/open-cluster-management-io.github.io: content/en/concepts/placement.md

group the remaining clusters into groups with a maximum of 150 clusters each.

```yaml
apiVersion: cluster.open-cluster-management.io/v1beta1
kind: Placement
metadata:
  name: placement1
  namespace: default
spec:
  clusterSets:
    - global
  decisionStrategy:
    groupStrategy:
      clustersPerDecisionGroup: 150
      decisionGroups:
      - groupName: prod-canary-west
        groupClusterSelector:
          labelSelector:
            matchExpressions:
              - key: prod-canary-west
                operator: Exists
      - groupName: prod-canary-east
        groupClusterSelector:
          labelSelector:
            matchExpressions:
              - key: prod-canary-east
                operator: Exists
```

The `decisionStrategy` section includes the following fields:
- `decisionGroups`: Represents a list of predefined groups to put decision results.
  Decision groups will be constructed based on the `decisionGroups` field at first. 
  The clusters not included in the `decisionGroups` will be divided to other decision 
  groups afterwards. Each decision group should not have the number of clusters 
  larger than the `clustersPerDecisionGroup`.

>   - `groupName`: Represens the name to be added as the value of label key `cluster.open-cluster-management.io/decision-group-name`

Reviewer: ```suggestion
  - `groupName`: Represents the name to be added as the value of label key `cluster.open-cluster-management.io/decision-group-name`
```

    of created `PlacementDecisions`.
  - `groupClusterSelector`: Defines the label selector to select clusters subset by label.
- `clustersPerDecisionGroup`: A specific number or percentage of the total selected 

>   clusters. The specific number will divide the placementDecisions to decisionGroups 

Reviewer: ```suggestion
  clusters. The specific number will divide the placementDecisions to decisionGroups,
```

>   each group has max number of clusters equal to that specific number.

Reviewer: ```suggestion
 the max number of clusters in each group equal  to that specific number.
```
